module/projectAttributes/helper.js

```diff
@@ -0,0 +1,221 @@
+				throw new Error(userRoleInformation)
+			}
+			let roleToAddForFilter = []
+			if (userRoleInformation.data.length > 0) {
+				userRoleInformation.data.map((eachResult) => {
+					roleToAddForFilter.push({
+						label: eachResult.title,
+						value: eachResult.code,
+					})
+				})
+			}
+			// Adding the roles into entities of projectAttributes
+			getProjectAttribute[1].entities = roleToAddForFilter
+			await projectAttributesQueries.createProjectAttributes(getProjectAttribute)
+			return {
+				success: true,
+				message: CONSTANTS.apiResponses.PROJECT_ATTRIBUTES_CREATED,
+			}
+		} catch (error) {
+			// If an error occurs, return an error response with status, message, and the error object
+			return {
+				status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+				message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+				errorObject: error,
+			}
+		}
+	}
+
+	/**
+	 * update project attributes
+	 * @method
+	 * @name update
+	 * @param {code} - which attributes need to be updated
+	 * @param {language} - language code for multilingual
+	 * @param {Object} bodyData  - reqBody
+	 * @returns {Object} .
+	 */
+
+	static async update(code, language, bodyData) {
+		try {
+			if (bodyData.translateData) {
+				// getting name and data from reqBody
+				const { name, data } = bodyData.translateData
+
+				// Fetch the project attributes document for the given code and language
+				const projectAttributesDocument = await projectAttributesQueries.projectAttributesDocument(
+					{
+						code: code,
+						[`translation.${language}`]: { $exists: true },
+					},
+					[`translation.${language}`]
+				)
+
+				if (projectAttributesDocument && projectAttributesDocument.length > 0) {
+					// Filter data to exclude already existing codes in the document
+					const existingCodes = projectAttributesDocument[0].translation[language].data || []
+					const filteredData = data.filter((eachValue) => !existingCodes.includes(eachValue.code))
+
+					if (filteredData.length > 0) {
+						// Update only if there is filtered data
+						await projectAttributesQueries.findAndUpdate(
+							{ code: code },
+							{ $set: { [`translation.${language}`]: { name, data: filteredData } } }
+						)
+					}
+				} else {
+					// If no document matches, upsert a new one
+					await projectAttributesQueries.findAndUpdate(
+						{ code: code },
+						{ $set: { [`translation.${language}`]: { name, data } } },
+						{ upsert: true }
+					)
+				}
+				return {
+					success: true,
+					message: CONSTANTS.apiResponses.PROJECT_ATTRIBUTES_UPDATED,
+				}
+			} else {
+				let projectAttributesDocument = await this.updateEntities('entities', code, bodyData.data)
+				if (projectAttributesDocument.success) {
+					return {
+						success: true,
+						message: CONSTANTS.apiResponses.PROJECT_ATTRIBUTES_UPDATED,
+					}
+				} else {
+					throw {
+						message: CONSTANTS.apiResponses.PROJECT_ATTRIBUTES_NOT_UPDATED,
```

Comment:
@praveenKDass when you throw an error make sure to add proper status as well like status: HTTP_STATUS_CODE.bad_request.status, else it will be always HTTP_STATUS_CODE.internal_server_error.status.
Please check all the places where you are returning errors, make sure it handled properly . and please  test it as well

Reply:
Yes @VISHNUDAS-tunerlabs  Handled the status and tested as well